packit/local_project.py

```diff
@@ -131,13 +132,23 @@ def refresh_the_arguments(self):
             )
 
     def _is_url(self, path_or_url):
+        try:
+            return _parse_url(path_or_url)
+        except:
+            logger.warning("path_or_url is nor directory nor url")
+        return False
+    
+    def _parse_url(self, path_or_url):
         try:
             res = requests.head(path_or_url)
```

Comment:
If we parse the URL, I wonder if we actually need this: once we know it's a URL, we should really treat it like that and not try doing requests - clone the repository instead.

Reply:
I don't fully understand, can you please clarify this please? Should I check wheter the url exists  or just that it is in the correct form?

Reply:
I may be asking too much to be honest, feel free to say no.

The way this code works now is that it is using the input value to determine whether it's a path or a URL by literally using it:
```python
if path_or_url:
    if os.path.isdir(path_or_url):
        working_dir = working_dir or path_or_url
    elif not offline and self._is_url(path_or_url):
        git_url = git_url or path_or_url
```

You have introduced a new method to parse the URL. My proposal is that if the URL is parsed correctly, we should then use it in the code instead of performing any requests which we are doing now.

But that's an ideal state. Feel free to complete the current pull request with the approach you chose and we can refactor in future. I don't want to make this too tough for you.